tests/test_symbol.py

```diff
@@ -205,3 +218,27 @@ def test_kind(self):
                     (ElfSymbol("foo", 0xFFFF0000, 1, elf_type, STB.GLOBAL),)
                 )
                 self.assertEqual(prog.symbol("foo").kind, drgn_kind)
+                symbols = prog.symbols("foo")
+                self.assertEqual(len(symbols), 1)
+                self.assertEqual(symbols[0].kind, drgn_kind)
+
+    def test_all_symbols(self):
+        elf_syms = (
+            (
+                ElfSymbol("two", 0xFFF11000, 1, STT.OBJECT, STB.LOCAL),
+                ElfSymbol("three", 0xFFF1200, 1, STT.OBJECT, STB.LOCAL),
+                ElfSymbol("one", 0xFFF10000, 1, STT.OBJECT, STB.GLOBAL),
+            ),
+            (
+                ElfSymbol("three", 0xFFF21000, 1, STT.OBJECT, STB.LOCAL),
+                ElfSymbol("two", 0xFFF20000, 1, STT.OBJECT, STB.GLOBAL),
+            ),
+            (ElfSymbol("three", 0xFFF30000, 1, STT.OBJECT, STB.GLOBAL),),
```

Comment:
The reason for the test failures on the 32-bit architectures is that `elf_symbol_program()` creates a section (filled with zeroes) to contain where the symbols would be. The size is `max_address - min_address`, which in this case is 0xFFF30000 - 0xFFF1200 = 0xeff3ee00, which is greater than `sys.maxsize` (2**31-1 on 32-bit systems). Come to think of it, this is silly and I should just put these in a `NOBITS` section, but in the meantime, just assign these closer values, e.g., `0xFFFFF0000`, `0xFFFF0008`, `0xFFFF0010`, etc.

Reply:
6732148a11e5362f624139fe967349a1cc5e0601 changed this to not actually write the data, but I think 0xFFF1200 is a typo anyways so you might as well change these to closer values.

Reply:
Wow, thanks for this catch. This is also why my test ran for ~6 seconds rather than instantly.